src/frequenz/dispatch/_bg_service.py

```diff
@@ -119,54 +145,53 @@ def new_lifecycle_events_receiver(self, type: str) -> Receiver[DispatchEvent]:
         )
 
     async def new_running_state_event_receiver(
-        self, type: str, *, unify_running_intervals: bool = True
+        self, type: str, *, merge_strategy: MergeStrategy
```

Comment:
I would handle the `NoMerge` case as a special case and add a `| None = None` in here. Besides having a (simpler[^1]) default, if we keep `NoMerge` as a regular strategy, we'll add a filter function to the receiver, and one that will iterate over all dispatches for no reason. IMHO the no merge case is special enough to deserve special treatment.

[^1]: I guess we can always also add `= NoMerge()` as default too.